packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -0,0 +1,213 @@
+import { monitor } from '@datadog/browser-core'
+import {
+  CreationReason,
+  IncrementalSource,
+  MouseMoveRecord,
+  Record,
+  RecordType,
+  SegmentContext,
+  SegmentMeta,
+} from '../types'
+
+export const MAX_SEGMENT_DURATION = 30_000
+export const MAX_MOUSE_MOVE_BATCH = 100
+
+export interface SegmentWriter {
+  write(data: string): void
+  complete(data: string, meta: SegmentMeta): void
+}
+
+// Segments are the main data structure for session replays.  They contain context information used
+// for indexing or UI needs, and a list of records (RRWeb 'events', renamed to avoid confusing
+// namings).  They are stored without any processing from the intake, and fetched one after the
+// other while a session is being replayed.  Their encoding (deflate) are carefully crafted to allow
+// concatenating multiple segments together.  Their approximative size limits how often they are
+// created have an impact on the replay.
```

Comment:
this sentence seems weird to me.
Also could you clarify the impact on the replay?